spawner.py

```diff
@@ -163,9 +167,34 @@ def update_required_context(suites):
     if total == 0:
         return
 
-    # let's link to the branch overview itself
-    url = 'https://github.com/%s/commits/%s' % (os.environ['github_repo'],
-                                                os.environ['github_branch'])
+    # OK, let's upload a very basic index file that just
+    # links to the results of all the required suites
+
+    results_suites = []
+    for i, suite in enumerate(suites):
+        name = suite['context']
+        with open("state/suite-%d/url" % i) as f:
+            url = f.read().strip()
+        result = (suite['rc'] == 0)
+        results_suites.append((name, result, url))
+
+    tpl_fname = os.path.join(dirname(realpath(__file__)),
+                             'utils', 'required-index.j2')
+
+    s3_key = '%s/%s/%s.%s/%s' % (os.environ['s3_prefix'],
+                                 os.environ['github_repo'],
+                                 os.environ['github_commit'],
+                                 random.randint(0, 2**15-1),
```

Comment:
Yeah, I understand why it's there, I mean why allow the possibility of overwriting?  A timestamp would be more reliably unique.